hugegraph-it/hg-it-minicluster/src/main/java/org/apache/hugegraph/it/node/AbstractNodeWrapper.java

```diff
@@ -0,0 +1,174 @@
+        this.port = 8086;
+    }
+
+    protected AbstractNodeWrapper(String host, int port, int clusterIndex, int cnt) {
+        this.host = host;
+        this.port = port;
+        this.clusterIndex = clusterIndex;
+        this.startTime = System.currentTimeMillis();
+        this.cnt = cnt;
+    }
+
+    /**
+     * Node Dir should be created before changing Config
+     */
+    @Override
+    public void createNodeDir() {
+        String destDir = getNodePath();
+        try {
+            try {
+                if (new File(destDir).exists()) {
+                    PathUtils.delete(Paths.get(destDir));
+                }
+            } catch (NoSuchFileException fileException) {
+                //no need to handle
+            }
+            // To avoid following symbolic links
+            try (Stream<Path> stream = Files.walk(Paths.get(TEMPLATE_NODE_DIR))) {
+                stream.forEach(
+                        source -> {
+                            Path destination =
+                                    Paths.get(destDir,
+                                              source.toString()
+                                                    .substring(TEMPLATE_NODE_DIR.length()));
+                            try {
+                                Files.copy(source,
+                                           destination,
+                                           LinkOption.NOFOLLOW_LINKS,
+                                           StandardCopyOption.COPY_ATTRIBUTES);
+                            } catch (IOException ioException) {
+                                LOG.error("Fail to copy files to node dest dir", ioException);
+                                throw new RuntimeException(ioException);
+                            }
+                        }
+                );
+            }
+        } catch (IOException ioException) {
+            LOG.error("Got error copying files to node dest dir", ioException);
+            throw new AssertionError();
+        }
+    }
+
+    @Override
+    public void createLogDir() {
+        String logPath = getLogPath();
+        try {
+            FileUtils.createParentDirectories(new File(logPath));
+        } catch (IOException e) {
+            LOG.error("Create log dir failed", e);
+            throw new AssertionError();
+        }
+    }
+
+    public void deleteDir() {
+        try {
+            PathUtils.deleteDirectory(Paths.get(getNodePath()));
+        } catch (IOException ex) {
+            try {
+                TimeUnit.SECONDS.sleep(1);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                ex.printStackTrace();
+                throw new AssertionError("Delete node dir failed. " + e);
+            }
+        }
+    }
+
+    /**
+     * @return (user.dir).target.id
+     */
+    public String getNodePath() {
+        return System.getProperty(USER_DIR) + File.separator + TARGET + File.separator +
```

Comment:
the method style should be dependency injection, as above talk, these config should come from clusterEnv